codegen.cpp

```diff
@@ -22,16 +22,28 @@ void NodeAdd::codegen() {
     lhs->codegen();
     cout << "  str x0, [sp, -16]!" << endl;
     rhs->codegen();
+    if(c_type->isPtrType()) {
+        cout << "  mov x1, 8" << endl;
+        cout << "  mul x0, x0, x1" << endl;
+    }
     cout << "  ldr x1, [sp], 16" << endl;
     cout << "  add x0, x0, x1" << endl;
 }
-
 void NodeSub::codegen(){
```

Comment:
Add comment explaining the 4 different cases because without that context, the if's are confusing. Same for `NodeAdd::codegen`. Missing implementation of num - ptr case, if that's supported which I still don't know.